File: internal/parser/typescript/testdata/sample.ts
// MaxRetries is the maximum number of retry attempts.
const MaxRetries = 3;

// Greeter interface defines objects that can greet.
interface Greeter {
  greet(name: string): string;
}

// Point is a 2D coordinate type.
type Point = { x: number; y: number };

// Animal is the base class for all animals.
class Animal {
  name: string;

  constructor(name: string) {
    this.name = name;
  }

  // speak outputs the animal's sound.
  speak(): void {
    console.log(this.name);
  }
}

// greet creates a greeting string for the given name.
function greet(name: string): string {
  return `Hello, ${name}`;
}

// helper is an unexported utility function.
function helper(x: number, y: number): number {
  return x + y;
}

File: internal/parser/python/testdata/sample.py
MAX_RETRIES = 3


class Animal:
    """Animal is the base class for all animals."""

    def __init__(self, name):
        """__init__ initializes a new Animal instance."""
        self.name = name

    def speak(self):
        """speak outputs the animal's sound."""
        print(self.name)


def greet(name):
    """greet creates a greeting string for the given name."""
    return f"Hello, {name}"


def helper(x, y):
    """helper adds two numbers."""
    return x + y
